src/pybind/mgr/rgw/module.py

```diff
@@ -146,18 +162,83 @@ def _cmd_rgw_realm_rm_zone_creds(self,
+            else:
+                keys = ['realm_name', 'realm_id', 'is_primary', 'endpoint', 'access_key', 'secret']
+                realm_token = RealmToken(**{k: realm_info[k] for k in keys})
+                realm_token_b = realm_token.to_json().encode('utf-8')
+                realm_token_s = base64.b64encode(realm_token_b).decode('utf-8')
+                realms_info.append({'realm': realm_info['realm_name'], 'token': realm_token_s})
+
+        return HandleCommandResult(retval=0, stdout=json.dumps(realms_info), stderr='')
+
+    @CLICommand('rgw zone update', perm='rw')
+    def update_zone_info(self, realm_name: str, zonegroup_name: str, zone_name: str, realm_token: str, endpoints: List[str]):
+        try:
+            retval, out, err = RGWAM(self.env).zone_modify(realm_name,
+                                                           zonegroup_name,
+                                                           zone_name,
+                                                           endpoints,
+                                                           realm_token)
+            return (retval, 'Zone updated successfully', '')
+        except RGWAMException as e:
+            self.log.error('cmd run exception: (%d) %s' % (e.retcode, e.message))
+            return (e.retcode, e.message, e.stderr)
+
     @CLICommand('rgw zone create', perm='rw')
     def _cmd_rgw_zone_create(self,
-                             realm_token : Optional[str] = None,
-                             zonegroup_name: Optional[str] = None,
                              zone_name: Optional[str] = None,
-                             endpoints: Optional[str] = None,
-                             start_radosgw: Optional[bool] = True):
+                             realm_token: Optional[str] = None,
+                             port: Optional[int] = None,
+                             placement: Optional[str] = None,
+                             start_radosgw: Optional[bool] = True,
+                             inbuf: Optional[str] = None):
         """Bootstrap new rgw zone that syncs with existing zone"""
+        try:
+            if inbuf:
+                rgw_specs = self._parse_rgw_specs(inbuf)
+            elif (zone_name and realm_token):
+                placement_spec = PlacementSpec.from_string(placement) if placement else None
+                rgw_specs = [RGWSpec(rgw_realm_token=realm_token,
+                                     rgw_zone=zone_name,
+                                     rgw_frontend_port=port,
+                                     placement=placement_spec)]
+            else:
+                return HandleCommandResult(retval=-errno.EINVAL, stdout='', stderr='Invalid arguments: -h or --help for usage')
+
+            for rgw_spec in rgw_specs:
+                retval, out, err = RGWAM(self.env).zone_create(rgw_spec, start_radosgw)
```

Comment:
I'm having a hard time understand the conditions that would have more than 1 rgw_spec. Could I get an example?

Reply:
@adk3798 the output is informative only. I fixed it to show all the created zones.
@alimaredia this support is something we normally provide on cephadm also so user can use the same yaml to create several entries (instead of having to run the cmd for each spec). For example, imagine the case when a user wants to create several secondary zones so he can provide a single Yaml spec that defines all of them.